internal/proxy/impl.go

```diff
@@ -1593,6 +1597,81 @@ func (node *Proxy) CalcDistance(ctx context.Context, request *milvuspb.CalcDista
 		return node.Retrieve(ctx, retrieveRequest)
 	}
 
+	// the vectors retrieved are random order, we need re-arrange the vectors by the order of input ids
+	arrangeFunc := func(ids *milvuspb.VectorIDs, retrievedFields []*schemapb.FieldData) (*schemapb.VectorField, error) {
+		var retrievedIds *schemapb.ScalarField
+		var retrievedVectors *schemapb.VectorField
+		for _, fieldData := range retrievedFields {
+			if fieldData.FieldName == ids.FieldName {
+				retrievedVectors = fieldData.GetVectors()
+				break
+			}
+			if fieldData.Type == schemapb.DataType_Int64 {
+				retrievedIds = fieldData.GetScalars()
+			}
+		}
+
+		if retrievedIds == nil || retrievedVectors == nil {
+			return nil, errors.New("Failed to fetch vectors")
+		}
+
+		dict := make(map[int64]int)
+		for index, id := range retrievedIds.GetLongData().Data {
+			dict[id] = index
+		}
+
+		inputIds := ids.IdArray.GetIntId().Data
+		if retrievedVectors.GetFloatVector() != nil {
+			floatArr := retrievedVectors.GetFloatVector().Data
+			element := retrievedVectors.GetDim()
+			result := make([]float32, 0)
```

Comment:
should we preset result size? this will avoid unnecessary copy when append happens.